packages/core/src/components/ozon-content/ozon-content.tsx

```diff
@@ -1,48 +1,199 @@
-import { h, Event, Component, ComponentInterface, Prop, Listen, EventEmitter } from '@stencil/core';
-
-import { ContentAnchor } from './ozon-content.interfaces';
-import { OzonContentTransformer } from './ozon-content.transformer';
+import {
+  Component,
+  ComponentInterface,
+  Event,
+  EventEmitter,
+  Fragment,
+  h,
+  Prop,
+} from "@stencil/core";
+import { ContentAnchor } from "./ozon-content.interfaces";
+import { v4 as uuid } from "uuid";
 
 @Component({
-  tag: 'dso-ozon-content',
-  styleUrl: 'ozon-content.scss'
+  tag: "dso-ozon-content",
+  styleUrl: "ozon-content.scss",
+  shadow: true,
 })
 export class OzonContent implements ComponentInterface {
   @Prop()
-  content!: string;
+  content: string | undefined;
 
   @Event()
   anchorClick!: EventEmitter<ContentAnchor>;
 
-  @Listen('click')
-  handleClick(event: PointerEvent) {
-    this.transformer.handleClickEvent(event);
-  }
+  getAttribute = (node: Node, name: string): string => {
+    return node instanceof Element ? node.getAttribute(name) ?? "" : "";
+  };
 
-  private container?: HTMLDivElement;
+  getLocalName = (node: Node) => {
+    return node instanceof Element ? node.localName : node.nodeName;
+  };
 
-  private transformer!: OzonContentTransformer;
+  childNodesToJsx = (node: Node, childNodeName?: string) => {
+    return Array.from(node.childNodes)
+      .filter((n) => !childNodeName || this.getLocalName(n) === childNodeName)
+      .map((n) => this.nodeToJsx(n));
+  };
 
-  componentWillLoad() {
-    this.transformer = new OzonContentTransformer(this.anchorClick)
-  }
+  nodeToJsx = (node: Node): JSX.Element | JSX.Element[] => {
+    switch (this.getLocalName(node)) {
+      default:
```

Comment:
Tsja, we hebben er geen regel over maar vinden we allemaal niet dat `default` statements in switches als laatste moeten?

Reply:
And down it goes.

Reply:
🛗